Sources/Commands/SwiftTool.swift

```diff
@@ -263,6 +263,7 @@ extension SwiftCommand {
                     additionalFileRules: FileRuleDescription.swiftpmFileTypes,
                     pluginScriptRunner: swiftTool.getPluginScriptRunner(),
                     pluginWorkDirectory: try swiftTool.getActiveWorkspace().location.pluginWorkingDirectory,
+                    disableSandboxForPluginCommands: swiftTool.options.security.shouldDisableSandbox,
```

Comment:
are we missing a test?

Reply:
It's complicated™️

We have a test that's been added in https://github.com/apple/swift-package-manager/pull/4283, but it didn't trigger here for two reasons:
- We are running neither macOS 12 nor installing the concurrency compatibility libs on Swift CI, which leads to plugin tests being skipped on macOS and the sandboxing tests are macOS-only (rdar://100780895)
- Build commands generated by plugins are running in a sandbox that currently has blanket approval for `/private/tmp` which also happens to be where we're placing test examples during test execution, so the sandbox is more open than normally during tests. https://github.com/apple/swift-package-manager/pull/4307 aims to address that by giving individual plugins unique temporary directories

Reply:
FYI, I have rebased the PR and revived it.